### ISSUE TITLE
File upstream issues on untagged repos

### ISSUE THREAD

**Issue by purcell** [melpa/melpa#1786]
This will allow us to greatly increase the coverage of melpa-stable.

Ideally we would automate the identification of relevant repos, and the creation of the issues with some standard boilerplate text.

This relates to #1771.

**Comment by tarsius** [comment]
I've assigned this to myself as a way to keep track of it. This does not necessarily mean I will do something.

**Comment by tarsius** [comment]
Maybe we should require tags for newly added packages?

/cc @riscy @milkypostman @purcell

**Comment by milkypostman** [comment]
I think there is also the idea since we have non-gnu ropa now that maybe. MELPA should forego the stable package repo. 

I’m guessing that non-gnu suffers the same issues as marmalade did before: it still requires manual publishing of “stable” packages?

im fine with requiring tags but think it’ll likely cause frustration for this who, like me, don’t really have interest in managing releases. 

So I’m saying I defer to you and @riscy with comments 😄

**Comment by purcell** [comment]
> I’m guessing that non-gnu suffers the same issues as marmalade did before: it still requires manual publishing of “stable” packages?

No, I think there's some auto-publishing of new versions on the GNU archives as long as package metadata (headers) get updated.

**Comment by purcell** [comment]
I personally don't care about this issue — we're hardly going to get everyone to add tags now. Perhaps in our review guidelines or checklist we should mention tagging.

**Comment by tarsius** [comment]
Okay then lets leave it as is at least for now. (I was just glimpsing at the open issues and noticed this.)

**Comment by milkypostman** [comment]
> No, I think there's some auto-publishing of new versions on the GNU archives as long as package metadata (headers) get updated.

Huh, in that case it seems like we should just leave the stable business for the Non-GNU ELPA folks? 😈

**Comment by riscy** [comment]
> Huh, in that case it seems like we should just leave the stable business for the Non-GNU ELPA folks?

This is sort of a tangent, but I've seen several authors submitting packages to MELPA when they've already got the same package on ELPA -- and they do this so that their users can access the very latest versions via MELPA.

I think nobody here feels too strongly about this issue.  But if we *were* considering removing MELPA stable, how much time/effort/cost go into sustaining it?  (I personally spend no time at all.)  Maybe there's an argument to be made.

**Comment by milkypostman** [comment]
I think stable is just this thing that we keep around but is only for a niche market. Many packages don't have tags, there are users of stable but not as many as regular MELPA. It's extra code. It's extra runtime to build all packages. It's not a big deal but it's still maintenance overhead.

And now that non-gnu ELPA exists, it's kinda duplicate functionality. In the past we've had times where we said "don't put ELPA packages in MELPA" (although my recollection is unclear on the details). So by similar logic it seems odd to keep stable around.

MELPA clearly has a purpose, which you pointed out is being able to get updates to the latest versions. MELPA was always just a shim between VCS and package.el.

**Comment by tarsius** [comment]
> This is sort of a tangent, but I've seen several authors submitting packages to MELPA when they've already got the same package on ELPA -- and they do this so that their users can access the very latest versions via MELPA.

Actually variants of [Non]GNU ELPA exist that do offer rolling releases: https://elpa.nongnu.org/nongnu-devel/, https://elpa.gnu.org/devel/index.html, but they are not being advertised.

IMO until that changes, we should continue to accept packages into MELPA even if they are already available from another *ELPA.

 > So by similar logic it seems odd to keep stable around.

We don't recommend using MELPA stable but the people who have been doing it anyway likely want to continue doing so. I don't think the appearance of [Non]GNU ELPA changes anything in this regard because these archives don't offer all the packages that are available from MELPA (stable).

**Comment by minad** [comment]
As far as I understand the new releases.melpa.org channel uses either tags or extracts the version from the package header. @tarsius Can you confirm? One should still encourage package developers to tag releases, but the tags are not required anymore for MELPA to build release tarballs. Can this issue be closed?

**Comment by tarsius** [comment]
> @tarsius Can you confirm?

Yes.

But for a longer answer you will have to wait until after release day. [Tomorrow, the first of the month. I've got 17 packages to release, including magit (which hasn't seen a release in half a year) and --hurray-- package-build.]

I agree that now is a good time to revisit the issues you are bumping.

**Comment by tarsius** [comment]
So the conclusion is that we should *not* file upstream issues on untagged repos". Doing so would be futile. There are thousands of them, nobody wants to open all these issues, and many/most would never be acted on by upstream.

The new `releases` channel also makes this less important because it falls back to getting version information from the library header, and if that is missing too, it uses `0.0`.

Removing the `stable` channel (and consequently the new `releases` channel) is discussed elsewhere (#6656).

We have already decided elsewhere against proactively removing packages that are distributed on (Non)GNU ELPA. If the maintainer wants to do it, we don't object. Now that the version string formats are compatible (and the actual version strings are the same, provided upstream does not mark their releases inconsistently) there is one less reason to remove such packages.

**Comment by minad** [comment]
> We have already decided elsewhere against proactively removing packages that are distributed on (Non)GNU ELPA. If the maintainer wants to do it, we don't object. Now that the version string formats are compatible (and the actual version strings are the same, provided upstream does not mark their releases inconsistently) there is one less reason to remove such packages.

Personally I recommend to keep packages on both MELPA and ELPA if feasible, given that ELPA sometimes has a feel of "political instability" to it, with occasionally radical policy discussions on emacs-devel.

**Comment by tarsius** [comment]
Right. That's another important reason to keep one's own packages in MELPA. But if a maintainer wants to remove their package (e.g., in support of those political reasons), we accept that.

**Comment by minad** [comment]
Of course, yes. I agree.
